### DISCUSSION TITLE
Notes or a Description field for Macros

### DISCUSSION THREAD

**Discussion by GeNTooGeek** [WarmUpTill/SceneSwitcher#1088] [Ideas]
It would be nice to have a notes or description field for the macros, to be able to document my thought process when creating/editing my macros.  I don't get back to them often enough and I sometimes end up spending too long trying to figure out why I organized things they way I did.  This is especially true with how active the development of the plugin is, a great thing and I am not complaining :).  Many times I am trying to determine if there was a specific reason for my madness 6 months prior, or did I just not realize I could have done it differently, or maybe there is just a new way to do it.

**Comment by GeNTooGeek** [comment]
As I have been working more on my macros, I wonder if doing this as Comment conditions and actions would be better than an individual field.  That way we could add notes wherever in the macro makes the most sense.

Oh, and then could the Groups be allowed to define Comment Conditions, as well?

**Comment by WarmUpTill** [comment]
Hm .. I like the idea of adding some sort of notes section, but I am not really sure about adding this as some sort of dummy condition type.
I feel that might cause a lot of confusion.

How about tying some sort of notes dialog to the macro instead?
Maybe via the right-click context menu?

![Example](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/a70f2842-0aad-4590-b5c2-0121b45e00f4)
